rs/rosetta-api/tests/system_tests/test_cases/account_balances.rs

```diff
@@ -0,0 +1,218 @@
+                                    .unwrap();
+                                let from_account = Account {
+                                    owner: caller.clone().sender().unwrap(),
+                                    subaccount: approve_arg.from_subaccount,
+                                };
+                                (block_idx, from_account, approve_arg.spender)
+                            }
+                            LedgerEndpointArg::TransferArg(transfer_arg) => {
+                                let block_idx = caller_agent
+                                    .transfer(transfer_arg.clone())
+                                    .await
+                                    .unwrap()
+                                    .unwrap()
+                                    .0
+                                    .to_u64()
+                                    .unwrap();
+                                let from_account = Account {
+                                    owner: caller.clone().sender().unwrap(),
+                                    subaccount: transfer_arg.from_subaccount,
+                                };
+                                (block_idx, from_account, transfer_arg.to)
+                            }
+                        };
+
+                        // Store the current balance of the involved accounts and add them to the list of accounts if not already present
+                        let balance_acc1 = caller_agent
+                            .balance_of(account1, CallMode::Query)
+                            .await
+                            .unwrap();
+                        let balance_acc2 = caller_agent
+                            .balance_of(account2, CallMode::Query)
+                            .await
+                            .unwrap();
+                        account_balance_at_block_idx.insert((account1, block_idx), balance_acc1);
+                        account_balance_at_block_idx.insert((account2, block_idx), balance_acc2);
+                        involved_accounts.insert(account1);
+                        involved_accounts.insert(account2);
+                        block_indices.insert(block_idx);
+                    }
+
+                    let mut current_balances = HashMap::new();
+                    for account in involved_accounts.clone().into_iter() {
+                        current_balances.insert(account, Nat(BigUint::zero()));
+                    }
+
+                    let mut block_indices_iter = block_indices.into_iter().collect::<Vec<u64>>();
+                    block_indices_iter.sort();
+
+                    wait_for_rosetta_to_sync_up_to_block(
+                        &rosetta_testing_environment.rosetta_client,
+                        rosetta_testing_environment.network_identifier.clone(),
+                        *block_indices_iter.iter().last().unwrap(),
+                    )
+                    .await
+                    .unwrap();
```

Comment:
Waiting for rosetta to sync up to the last block index in the sorted `block_indices_iter` introduced in [this fmt commit](https://github.com/dfinity/ic/pull/1296/commits/384909ed327af630e02d4a8d30ca11dc838e09ff) seems to have fixed the flakiness discussed above. :tada: